# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/farahhussain159-create/flyrank-ml-week1/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

In [2]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


I'm working on the "Refresh / Content Opportunity Scoring" lane. This lane focuses on predicting which web pages are showing a declining performance trend, so the content team can prioritize them for refresh. I chose this lane because it ties directly to a concrete business action — deciding which pages to refresh — rather than being a purely exploratory analysis.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

In [3]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


The question: which pages should be prioritized for content refresh, based on signals that their search performance is declining? The content/SEO team acts on this recommendation by deciding where to spend limited refresh effort. The cost of a wrong call: a false positive wastes team effort refreshing a page that wasn't actually declining, while a false negative means a genuinely declining page gets missed and continues losing traffic/rankings without intervention.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [4]:
import pandas as pd

url = "https://raw.githubusercontent.com/farahhussain159-create/flyrank-ml-week1/main/data/raw/content_refresh_anonymized.csv"
df = pd.read_csv(url)

print("Total rows:", len(df))
print("Total columns:", len(df.columns))

# Derive the declining label from trend_direction
df['is_declining_label'] = (df['trend_direction'] == 'down').astype(int)

declining_pct = round(df['is_declining_label'].mean() * 100, 2)
print("Declining %:", declining_pct, "%")

print("\nTrend direction breakdown:")
print(df['trend_direction'].value_counts())


Total rows: 30000
Total columns: 44
Declining %: 54.21 %

Trend direction breakdown:
trend_direction
down      16262
stable     5962
up         4388
new        2236
flat       1152
Name: count, dtype: int64


A quick look at this working dataset shows 30,000 rows across 44 columns, drawn from the FlyRank ML Internship dataset. Of these, 54.21% of pages are labeled as declining (derived from trend_direction == "down"), with the remaining split across stable, up, new, and flat trends. This scale and the near-balanced split between declining and non-declining pages make the lane worth pursuing over the next 7 weeks — there's enough signal in both classes to train and validate a meaningful classifier.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

In [5]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


This work will be able to say what I observed: a directional signal correlating certain features (impressions, average position, content age) with declining trend labels, useful as decision support for prioritization. It will not be able to claim causal proof — I cannot say that refreshing a page causes traffic recovery, nor can I predict future Google ranking changes with certainty. The model output is a prioritization signal, not a guarantee.

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.